alphagov/paas-cf
# [#163558206] Fix setting of the vcap user password on VMs

What
----

This follows on from https://github.com/alphagov/paas-bootstrap/pull/243.

We were previously attempting to set the vcap password for the VMs via the cloud-config. This was actually being ignored as cloud-config doesn't support an `env` key.

This Instead takes the common vcap password generated in paas-bootstrap and applies it to the CF and Prometheus manifests.

I've also taken the opportunity to move some of the Prometheus manifest generation logic out of the pipeline, and back into the manifest generation script.

How to review
-------------

* First review and deploy https://github.com/alphagov/paas-bootstrap/pull/243 to your dev env.
* Code review (commit-by-commit may be easiest)
* Deploy this to your dev env.
* Verify you the vcap password is set correctly on VMs from both the CF and Prometheus deployment.
  * This can be done by `bosh ssh`ing onto a VM, and then switching to the vcap user, and then attempting to sudo. eg:
```
# bosh ssh diego-cell/0

<SNIP>

diego-cell:~$ sudo -iu vcap
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

diego-cell:~$ whoami
vcap
diego-cell:~$ sudo -i
[sudo] password for vcap: 
diego-cell:~# whoami
root
diego-cell:~#
```

# :rotating_light: Before merging :rotating_light: 

Ensure that https://github.com/alphagov/paas-bootstrap/pull/243 has been merged and deployed to all persistent environments.

Who can review
--------------

Not me.
